mlrun/db/httpdb.py

```diff
@@ -3267,6 +3268,24 @@ def list_api_gateways(self, project=None):
         resp = self.api_call("GET", endpoint_path, error)
         return resp.json()
 
+    def create_api_gateway(
+        self,
+        project: str,
+        name: str,
+        api_gateway: mlrun.common.schemas.APIGateway,
```

Comment:
Note that now you have 2 schemas/representations for the `APIGateway` class - the one in the `schemas` which you reference here but it usually used in server-side code, and the one in `runtimes/api_gateway.py` which is the client-side code.
Since `httpdb` is client-side code, it's actually better to get the client-side object here. For full coverage you can do a `Union` of both classes and handle their to-dict serialization based on the class type. See `create_project` or `create_feature_set` for examples.

Reply:
@theSaarco Make sense! I've moved all validation logic under ApiGateway (client) class